src/main.rs

```diff
@@ -792,51 +779,49 @@ impl ManParser for Type4 {
         manpage.contains(".SH FUNCTION LETTERS")
     }
 
-    fn parse_man_page(&mut self, manpage: &str) -> Option<String> {
-        unimplemented!();
+    fn parse_man_page(&self, manpage: &str, cmdname: &str) -> Option<String> {
+        let options_section_re = regex!(r"\.SH FUNCTION LETTERS((?s:.)*?)(\.SH|\\Z)");
+        let options_section_matched = options_section_re.captures(manpage);
+        let mut options_section = options_section_matched.unwrap().get(1).unwrap().as_str();
+
+        let options_parts_re = regex!(r"\.TP((?s:.)*?)\.TP");
+        let mut options_matched = options_parts_re.captures(options_section);
+        // add_diagnostic(format!("Command is {}", cmdname));
+
+        if options_matched.is_none() {
+            // add_diagnostic("Unable to find options section");
+            return None;
+        }
+
+        let mut completions = Completions::new(cmdname);
+        while let Some(mat) = options_matched {
+            let data = mat.get(1).unwrap().as_str();
+            let data = remove_groff_formatting(data);
+            let data = data.trim().splitn(2, '\n').next_tuple::<(_, _)>();
+
+            if let Some(data) = data {
+                let option_name = data.0.trim();
+                if option_name.contains('-') {
+                    let option_name = unquote_double_quotes(option_name);
+                    let option_name = unquote_single_quotes(option_name);
+                    let option_desc = data.1.trim().replace('\n', " ");
+                    completions.add(option_name, &option_desc);
```

Comment:
Mimicking the pattern-match pattern mentioned elsewhere.

```suggestion
            if let Some((option_name, option_desc)) = data.trim().splitn(2, '\n').next_tuple() {
                let option_name = option_name.trim();
                if option_name.contains('-') {
                    let option_name = unquote_double_quotes(option_name);
                    let option_name = unquote_single_quotes(option_name);
                    let option_desc = option_desc.trim().replace('\n', " ");
                    completions.add(option_name, &option_desc);
```